src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -206,1025 +201,56 @@ def create_dir(directory):
-            start_date = end_date = acq_date_string
-            # build the USGS style S2-identifier
-            if utm_tile.startswith("T"):
-                utm_tile_base = utm_tile[1:]
-            bbox = get_bbox_from_S2_UTMtile(utm_tile_base)
-        else:
-            # get coordinate pairs from wkt string
-            str_1 = "POLYGON(("
-            str_2 = "))"
-            coords = area[area.find(str_1) + len(str_1) : area.rfind(str_2)].split(",")
-            # add one space to first pair for consistency
-            coords[0] = " " + coords[0]
-            lons = [float(pair.split(" ")[1]) for pair in coords]
-            lats = [float(pair.split(" ")[2]) for pair in coords]
-            bbox = (min(lons), min(lats), max(lons), max(lats))
-            start_date = start
-            end_date = end
-        usgs_args = {
-            "dataset": "SENTINEL_2A",
-            "bbox": bbox,
-            "start_date": start_date,
-            "end_date": end_date,
-        }
-        if clouds:
-            usgs_args["max_cloud_cover"] = clouds
-        if limit:
-            usgs_args["max_results"] = limit
-        scenes = self._api.search(**usgs_args)
-        self._api.logout()
-        if query:
-            # check if the UTM-Tile is correct, remove otherwise
-            for scene in scenes:
-                if scene["display_id"].split("_")[1] != utm_tile:
-                    scenes.remove(scene)
-            # remove redundant scene
-            if len(scenes) == 2:
-                for scene in scenes:
-                    prod_id = scene["display_id"].split("_")[-1]
-                    if prod_id != esa_prod_id:
-                        scenes.remove(scene)
-        if len(scenes) < 1:
-            gs.message(_("No product found"))
-            sys.exit()
-        scenes_df = pandas.DataFrame.from_dict(scenes)
-        if sortby:
-            # replace sortby keywords with USGS keywords
-            for idx, keyword in enumerate(sortby):
-                if keyword == "cloudcoverpercentage":
-                    sortby[idx] = "cloud_cover"
-                    # turn cloudcover to float to make it sortable
-                    scenes_df["cloud_cover"] = pandas.to_numeric(
-                        scenes_df["cloud_cover"]
-                    )
-                elif keyword == "ingestiondate":
-                    sortby[idx] = "acquisition_date"
-                # what does sorting by footprint mean
-                elif keyword == "footprint":
-                    sortby[idx] = "display_id"
-            self._products_df_sorted = scenes_df.sort_values(
-                sortby, ascending=[asc] * len(sortby), ignore_index=True
-            )
-        else:
-            self._products_df_sorted = scenes_df
-        gs.message(
-            _("{} Sentinel product(s) found").format(len(self._products_df_sorted))
-        )
+DATASOURCE_MAP = {
+    "ESA_CDSE": "cop_dataspace",
+    "GCS": "earth_search_gcs",
```

Comment:
In the case `"GCS": "earth_search_gcs"` I suppose google credentials are required to be configured via eodag? In the former version of `i.sentinel.download` this was not required because the GCS download URL was automatically composed by a given scene ID.
I think it is a great solution to use eodag as a one-stop-shop now, but if google credentials are required, this would also have to be indicated in the .html (or it is also possible to omit the Google cloud storage version entirely)

Reply:
Sorry for annoying so early - I noticed you only just began with this draft PR ;) I am looking forward to this solution!

Reply:
Hello @griembauer :D

> (or it is also possible to omit the Google cloud storage version entirely)

I think that would be a reasonable choice since Google cloud storage only provides one Sentinel product in the EODAG context (S2_MSI_L1C) which is already provided by Copernicus.

Reply:
Thanks @griembauer, it is correct to add to documentation since it is an important change